test/unit/org/apache/cassandra/transport/AuthMessageSizeLimitTest.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    @SuppressWarnings({"resource", "SameParameterValue"})
+    private SimpleClient client()
+    {
+        try
+        {
+            return SimpleClient.builder(nativeAddr.getHostAddress(), nativePort)
+                               .protocolVersion(ProtocolVersion.V5)
+                               .useBeta()
+                               .build()
+                               .connect(false, false);
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException("Error initializing client", e);
+        }
+    }
+
+    @Test
+    public void sendSmallAuthMessage()
+    {
+        doTest((client) ->
+               {
+                   AuthResponse authResponse = createAuthMessage("cassandra", "cassandra");
+                   client.execute(authResponse);
+
+                   QueryMessage createTableMessage = new QueryMessage("CREATE TABLE test_table (pk int PRIMARY KEY, v text)",
+                                                                V5_DEFAULT_OPTIONS);
+                   client.execute(createTableMessage);
+
+                   int valueLessThanMessageMaxSize = MAX_CQL_MESSAGE_SIZE - 500;
+                   QueryMessage queryMessage = createQueryMessage(valueLessThanMessageMaxSize);
+                   client.execute(queryMessage);
+               }
+        );
+    }
+
+    @Test
+    public void sendTooBigAuthMultiFrameMessage()
+    {
+        doTest((client) ->
+               {
+                   AuthResponse authResponse = createAuthMessage("cassandra", createIncorrectLongPassword(TOO_BIG_MULTI_FRAME_AUTH_MESSAGE_SIZE));
+                   try
+                   {
+                       client.execute(authResponse);
+                   } catch (RuntimeException e) {
+                       // ProtocolException: Auth CQL Message of size 262159 bytes exceeds allowed maximum of 130072 bytes
+                       Assert.assertTrue(e.getCause() instanceof ProtocolException);
+                   }
+                   Util.spinAssertEquals(false, () -> client.connection.channel().isOpen(), 10);
+
+               }
+        );
+    }
+
+    private void doTest(TestLogic testLogic)
+    {
+        try (SimpleClient client = client())
+        {
+            testLogic.run(client);
+        }
+    }
+    private interface TestLogic
+    {
+        void run(SimpleClient simpleClient);
+    }
+
+    private QueryMessage createQueryMessage(int valueSize)
+    {
+        StringBuilder query = new StringBuilder("INSERT INTO test_table (pk, v) VALUES (1, '");
+        for (int i = 0; i < valueSize; i++)
+            query.append('a');
+        query.append("')");
+        return new QueryMessage(query.toString(), V5_DEFAULT_OPTIONS);
+    }
+
+    private static String createIncorrectLongPassword(int length) {
+        StringBuilder password = new StringBuilder(length);
+        for (int i = 0; i < length; i++)
+            password.append('a');
+        return password.toString();
```

Comment:
Fun thing I just learned: You can replace all this with:

```
return "a".repeat(length);
```

(Appears above and in `MessageSizeLimitTest` too.)

Reply:
yes, it is a useful trick, unfortunately the method has been added only in Java 11 (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#repeat(int)), so I cannot use it for Cassandra 4.x